tel2puml/__main__.py

```diff
@@ -18,67 +18,220 @@
     help="Path to folder containing job json files",
-    default=".",
     dest="folder_path",
+    required=False,
 )
 
-parser.add_argument(
-    "-o",
-    "--output",
-    metavar="file",
-    help="Output file path",
-    default="./default.puml",
-    dest="puml_file_path",
+pv_exclusive_group.add_argument(
+    "file_paths",
+    nargs="*",
+    help="Input .json files containing job data",
+    default=[],
 )
 
-parser.add_argument(
-    "-sn",
-    "--sequence-name",
+pv_to_puml_parser.add_argument(
+    "-jn",
+    "--job-name",
     metavar="name",
-    help="Name given to the puml sequence diagram",
+    help=(
+        "Name given to the puml sequence diagram and prefix for the output "
+        "puml file"
+    ),
     default="default_name",
-    dest="puml_name",
+    dest="job_name",
 )
 
-parser.add_argument(
-    "file_paths",
-    nargs="*",
-    help="Input .json files containing job data",
-)
-
-parser.add_argument(
+pv_to_puml_parser.add_argument(
     "-group-by-job",
     help="Group events by job ID",
     action="store_true",
     dest="group_by_job",
 )
 
 
+def generate_config(file_path: str) -> Any:
+    """Parse config file.
+
+    :param file_path: The filepath to the config file
+    :type file_path: `str`
+    :return: Config file represented as a dictionary
+    :rtype: `dict`[`str`, `Any`]
+    """
+    with open(file_path, "r") as file:
+        return yaml.safe_load(file)
+
+
+def find_json_files(directory: str) -> list[str]:
+    """Walk a directory path and extract .json files.
+
+    :param directory: The directory path
+    :type directory: `str`
+    :return: List of filepaths
+    :rtype: `list`[`str`]
+    """
+    json_files = []
+
+    for root, _, files in os.walk(directory):
+        for file in files:
+            if file.endswith(".json"):
+                json_files.append(os.path.join(root, file))
+
+    if not json_files:
+        raise FileNotFoundError(
+            f"No .json files were found in directory '{directory}'"
+        )
+
+    return json_files
+
+
+def generate_component_options(
+    command: Literal["otel2puml", "otel2pv", "pv2puml"],
+    args_dict: dict[str, Any],
+) -> tuple[OtelPVOptions | None, PVPumlOptions | None]:
+    """Generate puml options objects based on CLI arguments.
+
+    :param command: The CLI command
+    :type command: `Literal`["otel2puml", "otel2pv", "pv2puml"]
+    :return: A tuple containing component options
+    :rtype: `tuple`[:class:`OtelPVOptions` | `None`, :class:`PVPumlOptions`
+    | `None`]
+    """
+
+    otel_pv_options, pv_puml_options = None, None
+    if command == "otel2puml" or command == "otel2pv":
+        OtelToPVArgs(**args_dict)
+        otel_pv_options = OtelPVOptions(
+            config=load_config_from_dict(
+                generate_config(args_dict["config_file"])
+            ),
+            ingest_data=args_dict["ingest_data"],
+        )
+    elif command == "pv2puml":
+        PvToPumlArgs(**args_dict)
```

Comment:
So here we should set this to a variable and use the values from the object to set the options dict rather than the args dict